tools/autograd/gen_inplace_or_view_type.py

```diff
@@ -370,6 +371,12 @@ def emit_view_lambda(f: NativeFunction, unpacked_bindings: List[Binding]) -> str
             # This is temporary. Nested tensors will be migrating to use SymInts and
             # nested_size and nested_strides will no longer be tensors.
             updated_unpacked_args.append(arg[:-1])
+        elif (arg == "jagged_offsets_") and arg_type == ConstRefCType(
+            BaseCType(tensorT)
+        ):
+            # Jagged offsets must be kept on the GPU, to prevent sync so cannot be passed
```

Comment:
I think follow up work would be to move _nested_tensor_size to the GPU in order to prevent the sync mentioned here (?)

Reply:
I think we'd need to move _nested_tensor_offset onto GPU, but size and stride can stay on CPU.

Reply:
hm but i'm iterating over the values from jagged_offsets (on GPU) to fill sizes a CPU tensor here, would that cause a sync?

Reply:
Yep, it definitely would. Another data point that treating the memory on the GPU like a cached value with the ground truth being readily available on the CPu might be a useful construct.

Reply:
what does it mean for the metadata on the GPU to be a cached value with the ground truth being readily available on the CPU?

Reply:
Do you want me to implement this before landing this PR?

Reply:
No, it's ok if this incur a sync for now. But we should track and discuss this.

Reply:
Great, tracking here #95815